src/braket/aws/aws_device.py

```diff
@@ -274,6 +275,69 @@ def arn(self) -> str:
         """str: Return the ARN of the device"""
         return self._arn
 
+    @property
+    def is_available(self) -> bool:
+        """bool: Return if the device is currently available"""
+        if self.status != "ONLINE":
+            return False
+
+        is_available_result = False
+
+        current_datetime_utc = datetime.utcnow()
+        for execution_window in self.properties.service.executionWindows:
+            weekday = current_datetime_utc.weekday()
+            current_time_utc = current_datetime_utc.time()
+
+            if (
+                execution_window.windowEndHour < execution_window.windowStartHour
+                and current_time_utc < execution_window.windowEndHour
+            ):
+                weekday = (weekday - 1) % 7
+
+            matched_day = execution_window.executionDay == ExecutionDay.EVERYDAY
+            matched_day = matched_day or (
+                execution_window.executionDay == ExecutionDay.WEEKDAYS and weekday < 5
+            )
+            matched_day = matched_day or (
+                execution_window.executionDay == ExecutionDay.WEEKENDS and weekday > 4
+            )
+            ordered_days = (
+                ExecutionDay.MONDAY,
+                ExecutionDay.TUESDAY,
+                ExecutionDay.WEDNESDAY,
+                ExecutionDay.THURSDAY,
+                ExecutionDay.FRIDAY,
+                ExecutionDay.SATURDAY,
+                ExecutionDay.SUNDAY,
+            )
+            matched_day = matched_day or (
+                execution_window.executionDay in ordered_days
+                and ordered_days.index(execution_window.executionDay) == weekday
+            )
+
+            matched_time = execution_window.windowStartHour == time(
+                0, 0
+            ) and execution_window.windowEndHour == time(23, 59, 59)
```

Comment:
Is this covered by the following check on #322-#327 ?

Reply:
This code was actually handling an all-day available device in the second after 23:59:59. I have improved this logic by changing line 289 as follows:

https://github.com/surfkansas/amazon-braket-sdk-python/blob/f27e1ce9d563137362714a0602b5e1ca7c7280d5/src/braket/aws/aws_device.py#L289